### ISSUE TITLE
Issues with windows installation

### ISSUE THREAD

**Issue by athype** [ChetanXpro/nodejs-whisper#130]
I've been battling several issues as I'm trying to make this work.
The below mentioned console output is my latest production, I have make and MinGW installed and recognized on PATH as the respective version commands work

```js
> node src/index.js

autoDownloadModelName tiny.en
options {
  modelName: 'tiny.en',
  autoDownloadModelName: 'tiny.en',
  verbose: false,
  removeWavFileAfterTranscription: false,
  withCuda: false,
  options: {
    outputInCsv: false,
    outputInJson: false,
    outputInJsonFull: true,
    outputInLrc: false,
    outputInSrt: false,
    outputInText: false,
    outputInVtt: false,
    outputInWords: false,
    translateToEnglish: false,
    wordTimestamps: false,
    timestamps_length: 20,
    splitOnWord: true
  }
}
code--- 1
stdout---
stderr--- '.' is not recognized as an internal or external command,
operable program or batch file.

[Nodejs-whisper] Error: '.' is not recognized as an internal or external command,
operable program or batch file.

[Nodejs-whisper] Error: '.' is not recognized as an internal or external command,
operable program or batch file.

[Nodejs-whisper] Error during processing: '.' is not recognized as an internal or external command,
operable program or batch file.

Error: Operation failed: '.' is not recognized as an internal or external command,
operable program or batch file.
```

**Comment by ChetanXpro** [comment]
@athype Yeah, I can see where these errors might be coming from. This package is not well tested on Windows. I will make this stable for Windows as well; this must be a small fix.

**Comment by athype** [comment]
@ChetanXpro I got to this result after going into WSL, going into the whisper.cpp directory within node_modules and running make, prior to that I was getting issues with make just not even running.

I really appreciate the quick response!

**Comment by AirplanegoBrr** [comment]
I got the example working with some tweaks

1. When building, this script expects the build to be outputted at the root of whisper.cpp dir, that is wrong, the built file is saved to `whisper.cpp\build\bin\Release`
2. `main.exe` is no longer supported by `whisper.cpp`, `whisper-cli.exe` is preferred
3. Windows, doesn't like to use `./` paths to run files, aka `./main` will not work. `main` will

So knowing all of this, I moved `whisper-cli.exe` and `whisper.dll` (required for the cli) to the `whisper.cpp` root dir

I renamed the cli exe to `main.exe`, I also made a copy of `main.exe` and renamed it to `main` (no exe) this fixes a check that the script does by checking for a `main` file in the root dir of `whisper.cpp`

and in the `whisperShell` function in `whisper.ts` I added `command = command.replace("./", "")`

Now the example script runs just fine!

# What to do next?

1. If the script is being run on Windows use the `whisper.cpp\build\bin\Release` dir not the root
2. prefer to use the `whisper-cli.exe` over the `main.exe`
3. Remove the need of using `./` when running the whisper exe

**Maybe my build environment is wrong and that's why I'm seeing the exe at `build\bin\Release`, I'm not sure**

I would expect most builds (on most OSes) to be in some sort of build directory. I'm not sure why this assumes that the build will be at the root. Maybe I need to dig more into the code more? I'm not sure.

I'd also like to see a check for `cmake` over `make` as the official docs for `whisper.cpp` use cmake. (however, this is just a personal opinion, and also is easier to install, IMO)

PLUS (in all the environments I've used on windows) You can't set a variable at the start of a command, `WHISPER_CUDA=1 make -j` fails on windows, it would be nice to see the official way of adding cuda by doing `cmake -B build -DGGML_CUDA=1` (this asumes that you are in the root of `whisper.cpp`)

I don't trust my skills enough to make a PR that would follow this projects cleanness and tidiness when it comes to the code, if needed I can run any Windows testing code to make sure it works, also I'd hate to break compatibility for other systems


Here is a video
(ignore the jump cut, windows is the best OS ever made)

https://github.com/user-attachments/assets/759af52b-201d-40ec-91b0-1b56b85abcd4

**Comment by ChetanXpro** [comment]
Hey @AirplanegoBrr, I recently updated whisper-cpp to the latest version in this project. that also introduced some issues. thanks for the detailed explanation - I will fix this issue today.

**Comment by ChetanXpro** [comment]
@AirplanegoBrr Could you please test the latest package? That issue should be fixed in #146

**Comment by AirplanegoBrr** [comment]
I just tried it, works great! Thank you for fixing it quickly! Will 100% be using this in my projects

**Comment by ChetanXpro** [comment]
Perfect!
